Update Go dependencies

awly opened the pull request:
Several dependencies can't be updated due to breakages (etcd and grpc
for example).

Also updated `ttlmap` usage since their API changed.

russjones commented:
@awly Bumping `ttlmap` is a little concerning because I think there were some internal changes in behavior? @fspmarshall I think you mentioned some to me?

Commit:
Update Go dependencies

Several dependencies can't be updated due to breakages (etcd and grpc
for example).

Also updated ttlmap usage since their API changed.

awly commented:
@russjones yeah, looks like https://github.com/gravitational/ttlmap/pull/2 at least dropped the `.Remove` method, which doesn't exist in upstream `ttlmap`.
I reverted that version bump.

awly commented:
The race in `logrus` was fixed in https://github.com/sirupsen/logrus/pull/1234
Let me bump that dep to `master`...

Commit:
Move all utils.InitLoggerForTests calls to TestMain

This prevents data races between changing the standard logger and it
acutally being used.

russjones commented on go.mod:
```diff
@@ -3,107 +3,118 @@ module github.com/gravitational/teleport
-	github.com/mattn/go-runewidth v0.0.4 // indirect
+	github.com/mailgun/minheap v0.0.0-20170619185613-3dbe6c6bf55f // indirect
+	github.com/mailgun/timetools v0.0.0-20170619190023-f3a7b8ffff47
+	github.com/mailgun/ttlmap v0.0.0-20170619185759-c1c17f74874f
```
Isn't this master from Mailgun, I thought we were at `91fd36b9004c1ee5a778739752ea1aba5638c03a` from `github.com/gravitational/ttlmap`?

awly replied:
`github.com/mailgun/ttlmap` is used in a few places instead of our fork.
let me update them, should be low-risk

awly replied:
Ugh, the mailgun stuff is tightly woven into `lib/limiter` and `vendor/github.com/gravitational/oxy/ratelimit/tokenlimiter.go` so we can't get rid of it from our deps entirely.

Commit:
u2f: use gravitataional/ttlmap instead of mailgun/ttlmap
